include/cuco/detail/static_set/static_set.inl

```diff
@@ -39,18 +40,19 @@ namespace cuco {
 namespace experimental {
 
 template <class Key,
-          class Extent,
           cuda::thread_scope Scope,
+          class Extent,
           class KeyEqual,
           class ProbingScheme,
           class Allocator,
           class Storage>
-constexpr static_set<Key, Extent, Scope, KeyEqual, ProbingScheme, Allocator, Storage>::static_set(
+constexpr static_set<Key, Scope, Extent, KeyEqual, ProbingScheme, Allocator, Storage>::static_set(
   Extent capacity,
   empty_key<Key> empty_key_sentinel,
   KeyEqual pred,
   ProbingScheme const& probing_scheme,
   Allocator const& alloc,
+  [[maybe_unused]] Storage const& storage,
```

Comment:
I don't like this. We're using `storage` as a dummy object here to deduce the `Storage` type from. A user might misinterpret that as an opportunity to pass in an existing storage for the container to use.

Reply:
I might have a solution in might that eliminates the `storage` argument from the ctor. I'm still fiddling with it a bit.